module-sys/Service/Service.cpp

```diff
@@ -21,35 +21,29 @@
 #include <typeinfo>                // for type_info
 #include <system/Constants.hpp>
 
-#if (DEBUG_SERVICE_MESSAGES > 0)
-#include <cxxabi.h>
-#endif
-
 // this could use Scoped() class from utils to print execution time too
 void debug_msg(sys::Service *srvc, const sys::Message *ptr)
 {
 #if (DEBUG_SERVICE_MESSAGES > 0)
 
-    int status           = -4; /// assign error number which is not  defined for __cxa_demangle
-    const char *realname = nullptr;
-    realname             = typeid(*ptr).name();
-    char *demangled      = abi::__cxa_demangle(realname, 0, 0, &status);
+    const char *realname = typeid(*ptr).name();
 
     assert(srvc);
     assert(ptr);
 
-    LOG_DEBUG("Handle message ([%s] -> [%s] (%s) data: %s %s",
+    if (xPortIsInsideInterrupt()) {
+        throw std::runtime_error("message sent from irq");
+    }
+    if (xTaskGetCurrentTaskHandle() == nullptr) {
```

Comment:
the debug messages being sent from interrupts/crit sections have been observed in case of this issue or it is just a general "side" improvement?

Reply:
it's just a minor improvement :) we cant log in irq, nor we can allocate - therefore I've added it.
We should never ever need these two if's.